arangodb-net-standard/ViewApi/ViewsApiClient.cs

```diff
@@ -72,15 +72,18 @@ public ViewApiClient(IApiClientTransport transport, IApiClientSerialization seri
         /// Create a new View
         /// POST /_api/view
         /// </summary>
-        /// <param name="body">The body of the request containing required properties.</param>
+        /// <param name="body">The body of the request containing required properties.</param>        
+        /// <param name="serializationOptions">Custom serialization options to be used.</param>
         /// <param name="token">A CancellationToken to observe while waiting for the task to complete or to cancel the task.</param>
         /// <returns></returns>
-        public virtual async Task<ViewResponse> PostCreateViewAsync(ViewDetails body,
-            CancellationToken token = default)
+        public virtual async Task<ViewResponse> PostCreateViewAsync(ViewDetails body, ApiClientSerializationOptions serializationOptions = null, CancellationToken token = default)
         {
             string uri = _apiPath;
-            var content = GetContent(body, new ApiClientSerializationOptions(true, true));
-            using (var response = await _transport.PostAsync(uri, content, token: token).ConfigureAwait(false))
+            var content = GetContent(body, 
+                serializationOptions ?? new ApiClientSerializationOptions(
```

Comment:
I don't think `serializationOptions` must be exposed to callers of this API method. ArangoDB Web API accepts a JSON with camelCase, we don't want callers to change the standard JSON that the library/driver sends to ArangoDB.

Instead we should turn on the option, like so:

```
var content = GetContent(body, new ApiClientSerializationOptions(
    useCamelCasePropertyNames: true,
    ignoreNullValues: true,
    applySerializationOptionsToObjectValuesInDictionaries: true));
```

Reply:
I just noticed that you asked about this in your previous comment: https://github.com/ArangoDB-Community/arangodb-net-standard/pull/445#issuecomment-1336861558

I missed that.

From what I understand, the option should be hardcoded by us in the API method because we don't want to allow other options to be modified.
One gotchat is that null properties provided in LinkProperties will be ignored, because our serialization options for `PostCreateViewAsync` ignore null values. Is that acceptable? I don't understand enough about views and fields to know if it is gonna be a problem.

I see the following mentioned on LinksProperties.IncludeAllFields:
```
        /// If set to true, then process all document attributes.
        /// Otherwise, only consider attributes mentioned in fields.
        /// Attributes not explicitly specified in fields will be
        /// processed with default link properties, i.e. null.
```

Reply:
> I don't think `serializationOptions` must be exposed to callers of this API method. ArangoDB Web API accepts a JSON with camelCase, we don't want callers to change the standard JSON that the library/driver sends to ArangoDB.
> 
> Instead we should turn on the option, like so:
> 
> ```
> var content = GetContent(body, new ApiClientSerializationOptions(
>     useCamelCasePropertyNames: true,
>     ignoreNullValues: true,
>     applySerializationOptionsToObjectValuesInDictionaries: true));
> ```

OK

Reply:
> I just noticed that you asked about this in your previous comment: [#445 (comment)](https://github.com/ArangoDB-Community/arangodb-net-standard/pull/445#issuecomment-1336861558)
> 
> I missed that.
> 
> From what I understand, the option should be hardcoded by us in the API method because we don't want to allow other options to be modified. One gotchat is that null properties provided in LinkProperties will be ignored, because our serialization options for `PostCreateViewAsync` ignore null values. Is that acceptable? I don't understand enough about views and fields to know if it is gonna be a problem.
> 
> I see the following mentioned on LinksProperties.IncludeAllFields:
> 
> ```
>         /// If set to true, then process all document attributes.
>         /// Otherwise, only consider attributes mentioned in fields.
>         /// Attributes not explicitly specified in fields will be
>         /// processed with default link properties, i.e. null.
> ```

If `IncludeAllFields` is set to `true`, then all document attributes are processed. The ones not specified in `Fields` are processed with a default value of `null`. 
1) When `IncludeAllFields == true` and `ignoreNullValues == true or false`, it's not a problem. `null` fields specified by the caller and removed by the serialization are replaced and processed as `null` by the server by default.
2) When `IncludeAllFields == false` and `ignoreNullValues == true`, it could cause a problem for `null` attributes specified by the caller, because the caller clearly intends to process the attributes as `null`, but the serialization will remove/ignore it.
3) When `IncludeAllFields == false` and `ignoreNullValues == false`, it will not cause any problem because `null` attributes specified by the caller will be processed as `null` as intended by the caller.
So, I think it makes sense to set `ignoreNullValues` to `false`. @DiscoPYF , your thoughts?

Reply:
Setting `ignoreNullValues` to `false`, causes a lot of errors with our existing tests because several property values are not specified. I think this will break a lot of existing code that are not specifying the whole view body. For better backward compatibility, I've added a new parameter `ignoreNullValuesOnSerialization` which defaults to `true`. @DiscoPYF , your thoughts please?

Reply:
I will take a closer look in the coming days. 👍

Reply:
I had another look and this approach could be used together with the parameter `ignoreNullValuesOnSerialization` so that it is only applied to links and fields. We could rename this parameter and do something like:

```c#
public virtual async Task<ViewResponse> PostCreateViewAsync(
    ViewDetails body,
    bool ignoreNullLinksAndFields = false,
    CancellationToken token = default)
{
    var content = GetContent(body, new ApiClientSerializationOptions(
        useCamelCasePropertyNames: true,
        ignoreNullValues: true,
        DictionaryOptions = new DictionaryValuesSerializationOptions
        {
            useCamelCasePropertyNames: true,
            ignoreNullValues: ignoreNullLinksAndFields
        }));
}
```

Not blocker, just a thought.